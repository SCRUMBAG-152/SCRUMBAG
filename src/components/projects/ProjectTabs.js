import React from 'react';
import PropTypes from 'prop-types';
import Paper from '@material-ui/core/Paper';
import { withStyles } from '@material-ui/core/styles';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import CommentIcon from '@material-ui/icons/Comment';
import SettingsIcon from '@material-ui/icons/Settings';
import BoardIcon from '@material-ui/icons/Dashboard';

const styles = {
  root: {
    flexGrow: 1,
    maxWidth: 500,
    justifyContent: 'center',
    display: 'flex',
    margin: '20px auto'

  },
};

export class ProjectTabs extends React.Component {

  handleChange= (e, value) =>{
      this.props.handleTabChange(e, value)
  }

  render() {
    const { classes } = this.props;

    return (
      <Paper square className={classes.root}>
        <Tabs
          value={this.props.tabValue}
          onChange={this.handleChange}
          variant="fullWidth"
          indicatorColor="secondary"
          textColor="secondary"
        >
          <Tab icon={<BoardIcon />} label="BOARD" />
          <Tab icon={<CommentIcon />} label="COMMENTS" />
          <Tab icon={<SettingsIcon />} label="SETTINGS" />
        </Tabs>
      </Paper>
    );
  }
}

ProjectTabs.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(ProjectTabs);