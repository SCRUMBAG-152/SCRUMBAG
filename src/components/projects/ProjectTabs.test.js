import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import { ProjectTabs } from './ProjectTabs';


describe('Project Tabs', () => {
    it('renders without crashing given the required props', () => {
      const props = {
        projectID: [], 
        project: [],
        auth: [], 
        classes: [], 
        cards: [], 
        columns: []
      }
      const wrapper = shallow(<ProjectTabs {...props} />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })
    

