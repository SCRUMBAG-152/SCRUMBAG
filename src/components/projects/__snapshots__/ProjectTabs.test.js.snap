// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Tabs renders without crashing given the required props 1`] = `
<WithStyles(Paper)
  square={true}
>
  <WithStyles(Tabs)
    indicatorColor="secondary"
    onChange={[Function]}
    textColor="secondary"
    variant="fullWidth"
  >
    <WithStyles(Tab)
      icon={<pure(DashboardIcon) />}
      label="BOARD"
    />
    <WithStyles(Tab)
      icon={<pure(CommentIcon) />}
      label="COMMENTS"
    />
    <WithStyles(Tab)
      icon={<pure(SettingsIcon) />}
      label="SETTINGS"
    />
  </WithStyles(Tabs)>
</WithStyles(Paper)>
`;
